source/base/timer.cc

```diff
@@ -37,6 +38,83 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+namespace internal
+{
+  namespace Timer
+  {
+    namespace
+    {
+      /**
+       * Type trait for checking whether or not something is a duration.
+       */
+      template <typename T>
+      struct is_duration : std::false_type {};
+
+      /**
+       * Specialization for durations.
+       */
+      template <typename Rep, typename Period>
+      struct is_duration<std::chrono::duration<Rep, Period>> : std::true_type {};
```

Comment:
If you only intent to check that the given type is in fact a `std::chrono::duration` type, I agree with this type trait.